### DISCUSSION TITLE
[Feature] Typesense API better env variables

### DISCUSSION THREAD

**Discussion by SuberSeb** [immich-app/immich#2046] [Feature Request]
### The feature

Hello.

I have a suggestion for improvement Typesense API env variables:

1) Make TYPESENSE_API_KEY optional if TYPESENSE_ENABLED set to false. Currently if I set false I need to fill TYPESENSE_API_KEY anyway even if I don't want to use it;
2) Add TYPESENSE_API_HOST and TYPESENSE_API_PORT. Currently is seems to be hardcoded to 'typesense' hostname. I tried to run that container as 'immich-typesense' but there are no option to set another hostname. So I got exceptions in logs and http 500 error when accessing immich web gui.

Thank you in advance.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
It should be optional when `TYPESENSE_ENABLED=false`
![image](https://user-images.githubusercontent.com/4334196/226809314-3feec130-04b9-48b2-a92d-ffda337c36f7.png)

There are variables for host, port, and protocol, though they are not currently documented. I think there is a PR to add them already.
![image](https://user-images.githubusercontent.com/4334196/226809360-28d10dcb-9a59-4fe6-b275-4cace865d2a4.png)

There is a change your 500 error is not actually related to either of these issues, but is caused by something else. Looking at the immich-server logs and the immich-microservice logs would probably provide some light to what is happening.

**Comment by SuberSeb** [comment]
As for 500 error, I had TYPESENSE_ENABLED=false and TYPESENSE_API_KEY empty. Only after I put let's say '123' to the TYPESENSE_API_KEY I had no more 500 error (exception in logs was about empty TYPESENSE_API_KEY but it should be ignore even if key empty I think).
